tiledb/libtiledb.pyx

```diff
@@ -504,13 +506,23 @@ def stats_reset():
     tiledb.core.init_stats()
 
 def stats_dump():
-    """Print all TileDB internal statistics values to standard output."""
-    tiledb_stats_dump(stdout)
+    """Return TileDB internal statistics as a string."""
+    fd, path = tempfile.mkstemp()
+    stats = ""
+    try:
+        dump_file = fdopen(fd, "w")
+        if tiledb_stats_dump(dump_file) == TILEDB_ERR:
```

Comment:
Would you mind to try using `tiledb_stats_dump_str` here, see
https://github.com/TileDB-Inc/TileDB/blob/8f0e60a2713abadd68ba7f7e86e0850b93d1b12c/tiledb/sm/c_api/tiledb.h#L5580-L5596

This will look something like:
```
    cdef char* stats_str_ptr = NULL
    if tiledb_stats_dump_str(&stats_str_ptr) == TILEDB_ERR:
      raise TileDBError("Failed to dump internal stats")

    stats_str = stats_str_ptr.decode('UTF-8', 'strict')
    free(stats_str_ptr)

    return stats_str
```

Reply:
Should I print or return `stats_str`? In the comment above, you we should print the output, but here you are returning the output.

Reply:
Still print at the end, just meant to highlight how to get the data back from the `_str` version.